prog/download_boot_image.rb

```diff
@@ -43,22 +43,17 @@ def custom_url
   end
 
   label def wait_learn_storage
-    reap.each do |st|
+    reaper = proc do |st|
```

Comment:
Do you have thoughts about lambda vs proc here? (re: how labels work). 

Also, it's somewhat rare-ish to see higher order programming as such in ruby outside of the block construct. It does happen when alternatives are exhausted.

Also don't copy my mistake of adding `?` to non-predicates. That was just a total oops it'd be nice to fix (shotgun surgery on the code base)

Reply:
Also, you may wonder...why did I do that weird thing for semaphores, turning what might have been an `if` into a block? Well, it was a half-finished thought, because, it is often useful to describe why a state transition is occurring, in plain text, for the log, and I thought putting it into a block would give me a bit more breathing room to compose functionality in.

But I never completed the thought, which was part of that rudimentary era in early April, when the first versions of Semaphores was written. So by modern standards (and even back then?), it'd be bad form to leave an incomplete thought that makes the code unusual.